packages/api/amplify_api/example/integration_test/graphql/api_key_test.dart

```diff
@@ -105,6 +105,66 @@ void main({bool useExistingTestUser = false}) {
             expect(blogFromEvent?.name, equals(name));
           });
 
+          testWidgets(
+              'subscribe() should handle multiple streams synchronously',
+              (WidgetTester tester) async {
+            final readyCompleter = Completer<void>();
+            final readyCompleter2 = Completer<void>();
+            final dataCompleter = Completer<Blog>();
+            final dataCompleter2 = Completer<Blog>();
+
+            final name =
+                'Integration Test Blog - subscription sync test ${uuid()}';
+
+            final subscriptionRequest1 = ModelSubscriptions.onCreate(
+              Blog.classType,
+              authorizationMode: APIAuthorizationType.apiKey,
+            );
+
+            final stream1 = Amplify.API.subscribe(
+              subscriptionRequest1,
+              onEstablished: expectAsync0(readyCompleter.complete),
+            );
+            stream1.listen(
+              ((event) {
+                if (event.data?.name == name) {
+                  final blogFromEvent = event.data;
+                  expect(blogFromEvent?.name, equals(name));
```

Comment:
nit: I think this is guaranteed to be true bc of the if statement it's inside. Is this added just so the test has an expect? If so, maybe this could be removed and the expect could be added to the final lines around the completer completing.

Reply:
Yah I think this copy-paste artifact is from another test. Removed.